clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerMetrics.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.producer.internals;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.kafka.common.MetricNameTemplate;
+import org.apache.kafka.common.metrics.Metrics;
+
+public class ProducerMetrics {
+
+    public SenderMetricsRegistry senderMetrics;
+
+    public ProducerMetrics(Set<String> tags, String metricGrpPrefix) {
```

Comment:
The second parameter is not needed?

Reply:
You're right, it's not used in this checkin. But it'll actually get used once I convert more metrics over to this technique. In particular, the SelectorMetrics are used in both the producer and consumer. Stuff like `request-rate` and `connection-close-rate`. When used in the producer, the MBean is something like `kafka.producer:type=producer-metrics` and when used in the consumer, the MBean is something like `kafka.consumer:type=consumer-metrics`. So this 2nd parameter is passed in and used to contruct the MBean name.